From ff2febc04a45d697885a939f71e0f2d6153f22aa Mon Sep 17 00:00:00 2001
Date: Thu, 27 Aug 2026 16:57:59 +0300
Subject: [PATCH] windows: skip elf-cleaner, pin GO_PLATFORM

---
 tunnel/tools/CMakeLists.txt | 14 ++------------
 1 file changed, 2 insertions(+), 12 deletions(-)

diff --git a/tunnel/tools/CMakeLists.txt b/tunnel/tools/CMakeLists.txt
index 8557c5ab..0e2a08b5 100644
--- a/tunnel/tools/CMakeLists.txt
+++ b/tunnel/tools/CMakeLists.txt
@@ -30,17 +30,7 @@ add_custom_target(libwg-go.so WORKING_DIRECTORY "${CMAKE_CURRENT_SOURCE_DIR}/lib
     TARGET=${CMAKE_C_COMPILER_TARGET}
     DESTDIR=${CMAKE_LIBRARY_OUTPUT_DIRECTORY}
     BUILDDIR=${CMAKE_LIBRARY_OUTPUT_DIRECTORY}/../generated-src
+    GO_PLATFORM=windows-amd64
 )
 
-# Strip unwanted ELF sections to prevent DT_FLAGS_1 warnings on old Android versions
-file(GLOB ELF_CLEANER_SOURCES elf-cleaner/*.c elf-cleaner/*.cpp)
-add_custom_target(elf-cleaner COMMENT "Building elf-cleaner" VERBATIM COMMAND cc
-        -O2 -DPACKAGE_NAME="elf-cleaner" -DPACKAGE_VERSION="" -DCOPYRIGHT=""
-        -o "${CMAKE_CURRENT_BINARY_DIR}/elf-cleaner" ${ELF_CLEANER_SOURCES}
-)
-add_custom_command(TARGET libwg.so POST_BUILD VERBATIM COMMAND "${CMAKE_CURRENT_BINARY_DIR}/elf-cleaner"
-        --api-level "${ANDROID_NATIVE_API_LEVEL}" "$<TARGET_FILE:libwg.so>")
-add_dependencies(libwg.so elf-cleaner)
-add_custom_command(TARGET libwg-quick.so POST_BUILD VERBATIM COMMAND "${CMAKE_CURRENT_BINARY_DIR}/elf-cleaner"
-        --api-level "${ANDROID_NATIVE_API_LEVEL}" "$<TARGET_FILE:libwg-quick.so>")
-add_dependencies(libwg-quick.so elf-cleaner)
+
-- 
2.44.0.windows.1

